File: src/main/java/io/kapaseker/kharcho/nodes/PseudoTextElement.java
package io.kapaseker.kharcho.nodes;

import io.kapaseker.kharcho.internal.QuietAppendable;
import io.kapaseker.kharcho.parser.Tag;

/**
 * Represents a {@link TextNode} as an {@link Element}, to enable text nodes to be selected with
 * the {@link io.kapaseker.kharcho.select.Selector} {@code :matchText} syntax.
 * @deprecated use {@link Element#selectNodes(String, Class)} instead, with selector of <code>::textnode</code> and class <code>TextNode</code>.
 */
@Deprecated
public class PseudoTextElement extends Element {

    public PseudoTextElement(Tag tag, String baseUri, Attributes attributes) {
        super(tag, baseUri, attributes);
    }

    @Override
    void outerHtmlHead(QuietAppendable accum, Document.OutputSettings out) {
    }

    @Override
    void outerHtmlTail(QuietAppendable accum, Document.OutputSettings out) {
    }
}

File: src/main/kotlin/io/kapaseker/kharcho/nodes/PseudoTextElement.kt
package io.kapaseker.kharcho.nodes

import io.kapaseker.kharcho.internal.QuietAppendable
import io.kapaseker.kharcho.parser.Tag

/**
 * Represents a [TextNode] as an [Element], to enable text nodes to be selected with
 * the [io.kapaseker.kharcho.select.Selector] `:matchText` syntax.
 */
@Deprecated("use {@link Element#selectNodes(String, Class)} instead, with selector of <code>::textnode</code> and class <code>TextNode</code>.")
class PseudoTextElement(tag: Tag?, baseUri: String?, attributes: Attributes?) :
    Element(tag, baseUri, attributes) {
    override fun outerHtmlHead(accum: QuietAppendable, out: Document.OutputSettings) {
    }

    override fun outerHtmlTail(accum: QuietAppendable, out: Document.OutputSettings) {
    }
}
